### PR TITLE
Add HSigmoidFusionWithClampDiv fusion and remove duplicated patterns …

### PR DESCRIPTION
**Pull request by mateusztabaka**
…in HSwishFusion

Ticket: 54937

Since HSwish is x * HSigmoid(x), we can remove some redundant HSwish transformations so they can be handled by x * HSigmoid(x) pattern.

### CODE REVIEW

**Review by mitruska — COMMENTED** [review]
As I understand, HSwish transformation patterns that can be covered by HSigmoid patterns are removed. 
Does it mean HSigmoid fusion should be always executed before the HSwish fusion?
https://github.com/openvinotoolkit/openvino/blob/6bb5a2c3cb132dd3c56748680b02837b2bc7f7d8/inference-engine/src/transformations/src/transformations/common_optimizations/common_optimizations.cpp#L111-L112

**Comment by mateusztabaka** [comment]
> Does it mean HSigmoid fusion should be always executed before the HSwish fusion?
Yes and it works that way right now. `common_fusions` is a set of matchers and they're not applied one after another, but rather any of them can be applied as soon as its pattern matches particular subgraph.

**Review by postrational — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]
@GlebKazantaev Please have a look.

File: inference-engine/src/transformations/src/transformations/common_optimizations/hsigmoid_fusion.cpp [hunk 1]
```diff
@@ -201,3 +204,49 @@ ngraph::pass::HSigmoidFusionWithClamp::HSigmoidFusionWithClamp() {
...
+
+ngraph::pass::HSigmoidFusionWithClampDiv::HSigmoidFusionWithClampDiv() {
+    MATCHER_SCOPE(HSigmoidFusionWithClampDiv);
+    // Replaces a sub-graph (Clamp(x + 3, 0, 6) / 6) with a HSigmoid op.
+    auto input = ngraph::pattern::any_input();
+    auto add_constant = ngraph::pattern::wrap_type<ngraph::opset4::Constant>();
+    auto add = ngraph::pattern::wrap_type<ngraph::opset4::Add>({input, add_constant});
+    auto clamp = ngraph::pattern::wrap_type<ngraph::op::v0::Clamp>({add});
+    auto div_constant = ngraph::pattern::wrap_type<ngraph::opset4::Constant>();
+    auto div = ngraph::pattern::wrap_type<ngraph::opset4::Divide>({clamp, div_constant});
```
Comment (GlebKazantaev) [inline inference-engine/src/transformations/src/transformations/common_optimizations/hsigmoid_fusion.cpp:218]: Use latest opset

File: inference-engine/src/transformations/src/transformations/common_optimizations/hsigmoid_fusion.cpp [hunk 2]
```diff
@@ -201,3 +204,49 @@ ngraph::pass::HSigmoidFusionWithClamp::HSigmoidFusionWithClamp() {
...
+
+    ngraph::matcher_pass_callback callback = [=](ngraph::pattern::Matcher &m) {
+        auto &pattern_to_output = m.get_pattern_value_map();
+        auto x_output = pattern_to_output.at(input);
+
+        auto add_const_value = std::dynamic_pointer_cast<ngraph::opset4::Constant>(pattern_to_output.at(add_constant).get_node_shared_ptr());
+        auto div_const_value = std::dynamic_pointer_cast<ngraph::opset4::Constant>(pattern_to_output.at(div_constant).get_node_shared_ptr());
```
Comment (GlebKazantaev) [inline inference-engine/src/transformations/src/transformations/common_optimizations/hsigmoid_fusion.cpp:225]: Check for nullptr

**Review by GlebKazantaev — CHANGES_REQUESTED** [review]

File: inference-engine/src/transformations/src/transformations/common_optimizations/hsigmoid_fusion.cpp (hunk 2 above)
Comment (mateusztabaka) [reply inference-engine/src/transformations/src/transformations/common_optimizations/hsigmoid_fusion.cpp]: it's done by has_constant_value function

**Review by GlebKazantaev — APPROVED** [review]

**Pull request merged**